staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -463,6 +507,8 @@ var string2token = map[string]Token{
 	"!=":    NotEqualsToken,
 	"notin": NotInToken,
 	"(":     OpenParToken,
+	"=*":    StartsWithToken,
```

Comment:
How about using "^\*" for StartsWithToken and "*$" for EndsWithToken?

Reply:
It's funny you mention that because initially I had `=$` and `=^`. 

`=*` and `*=` felt more natural than regex symbols. The operator feel more self documenting in this way.

Reply:
Are there any examples of `=*` and `*=` being used in other languages or systems as prefix and suffix matching represents?
As for these notations, I think it's better to follow the convention if possible.

I don't know your first idea because I couldn't find it, how about `^=` and `$=` ?
These notations are also used by the selectors for CSS.

`[att^=val]` Represents an element with the att attribute whose value begins with the prefix "val".
`[att$=val]` Represents an element with the att attribute whose value ends with the suffix "val".
ref: https://www.w3.org/TR/selectors-3/#attribute-substrings

Reply:
@t-inu +1

Reply:
@t-inu +1 for `^=` and `$=`

Reply:
I'm more than happy to change it, particularly if there's an established convention. I'll make the change before the upcoming SIG meeting.